src/core/sys/posix/sys/socket.d

```diff
@@ -594,14 +594,24 @@ version( CRuntime_Glibc )
 
     enum : uint
```

Comment:
Pedantically speaking, those values are operating system-specific, as well as libc-specifc. In this case they're the correct for linux/glibc, but not for hurd/glibc. Can you add `version (linux)` before the enum definition?

(I know that you didn't write the original code, but we try to improve things incrementally whenever the opportunity arises.)

Reply:
If they're specific to Linux, they're supposed to go in core.sys.linux.sys.socket, not core.sys.posix.sys.socket. Nothing should go in the posix modules unless they're standard POSIX.

Reply:
AFAICS, the latest edition (2016) of the Posix standard defines only the following constants:
```
MSG_CTRUNC
MSG_DONTROUTE
MSG_EOR
MSG_OOB
MSG_NOSIGNAL
MSG_PEEK
MSG_TRUNC
MSG_WAITALL
```
The values for those constants are implementation defined.
You may argue that, the rest of the values that are not part of the standard should be outside of the `core.sys.posix.` headers, but I think splitting the enum definition would be confusing and will do more harm than good.

Reply:
They're just constants. It's not a named enum, and the way that druntime's bindings are laid out, the Linux-specific stuff _must_ go under core.sys.linux and not core.sys.posix. That's how the bindings are organized. It's just plain wrong to put system-specific bindings in core.sys.posix. You do end up with version blocks in the modules in core.sys.posix, because stuff like the actual values of the constants change, but only stuff that's actually part of POSIX goes in core.sys.posix. That way, we don't have someone on Linux importing core.sys.posix and ending up with Linux-specific code (or someone on FreeBSD importing core.sys.posix and ending up with FreeBSD-specific code). The core.sys.linux.* modules publicly import their core.sys.posix counterparts so that someone on Linux can can just import the linux one and get everything if they don't care about portability, but the posix modules must not contain _anything_ that isn't POSIX.

Reply:
> unless CRuntime_Glibc always implies Linux.

It does:

https://github.com/dlang/dmd/blob/4c5bb35ffcfdcd29c9726c00f46c1f6a81c80a05/src/dmd/mars.d#L1212-L1217

Reply:
@jacob-carlborg the logic in dmd's code is the other way around: linux implies glibc. It's also not correct as there are other libc implementations for linux even for x86[_64] that dmd can support, notably musl libc.

Reply:
But `CRuntime_Glibc` is not set on any other platforms so what’s the difference.

Reply:
@jacob-carlborg obviously that's because we haven't suppoerted such platforms *yet* ;)